sui/deploy-gateway.js

```diff
@@ -107,6 +106,40 @@ async function processCommand(config, chain, options) {
     printInfo('Gateway deployed', JSON.stringify(contractConfig, null, 2));
 }
 
+async function upgradeGateway(chain, options, keypair, client) {
+    const contractsConfig = chain.contracts;
+    const packageName = 'axelar_gateway';
+    const builder = new TxBuilder(client);
+    const chainConfig = contractsConfig[packageName];
+    chainConfig.objects = chainConfig.objects || {};
+    chainConfig.objects.upgradeCap = options.upgradeCap || chainConfig.objects.UpgradeCap;
+
+    await upgradePackage(client, keypair, packageName, contractsConfig[packageName], builder, options);
+}
+
+async function processCommand(config, chain, options) {
+    const [keypair, client] = getWallet(chain, options);
+    await printWalletInfo(keypair, client, chain, options);
+
+    chain.contracts.axelar_gateway = chain.contracts.axelar_gateway ?? {};
+
+    if (!options.upgrade) {
+        await deployGateway(config, chain, options, keypair, client);
+    } else {
+        await upgradeGateway(chain, options, keypair, client);
+    }
+
+    if (options.offline) {
+        const { txFilePath } = options;
+        validateParameters({ isNonEmptyString: { txFilePath } });
+
+        const txB64Bytes = toB64(options.txBytes);
+
+        writeJSON({ status: 'PENDING', bytes: txB64Bytes }, txFilePath);
```

Comment:
file content fields should be consistent with EVM upgrade. What about the signed tx?

Reply:
Yeah, forgot that signing will need to be a separate step. Regardless, the combined signed tx can be stored within this same file, and we can mimic the following layout, instead of creating a separate file.
https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/evm/deploy-gateway-v6.2.x.js#L488-L493

This should also work for normal non-multisig accounts, since we might not use a multisig for everything